gallery_dl/extractor/2chen.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://2chen.moe/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class _2chenThreadExtractor(Extractor):
+    """Extractor for 2chen threads"""
+    category = "2chen"
+    subcategory = "thread"
+    directory_fmt = ("{category}", "{board}", "{thread} {title}")
+    filename_fmt = "{time} {filename}.{extension}"
+    archive_fmt = "{hash}"
+    root = "https://2chen.moe"
+    pattern = r"(?:https?://)?2chen\.moe/([^/?#]+)/(\d+)"
+    test = (
+        ("https://2chen.moe/jp/303786", {
+            "count": ">= 10",
+        }),
+    )
+
+    def __init__(self, match):
+        Extractor.__init__(self, match)
+        self.board, self.thread = match.groups()
+
+    def items(self):
+        url = "{}/{}/{}".format(self.root, self.board, self.thread)
+        page = self.request(url, encoding="utf-8").text
+        data = self.metadata(page)
+        yield Message.Directory, data
+        for post in self.posts(page):
+            if not post["url"]:
+                continue
+            post.update(data)
+            post["time"] = text.parse_int(post["date"].timestamp())
+            yield Message.Url, post["url"], text.nameext_from_url(
+                post["filename"], post)
+
+    def metadata(self, page):
+        title = text.extract(page, "<h3>", "</h3>")[0]
+        return {
+            "board" : self.board,
+            "thread": self.thread,
```

Comment:
`thread` is `"all"` for URLs like https://2chen.moe/all/303786. Try to extract the real name from HTML if possible.